### PR TITLE
RUN-4395/ Converted bounds_changed_state_tracker to typescript

### PR DESCRIPTION
**Pull request by pbaize**
Test results
[Win 10](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b75e3fc6107b45e6d1eba46)
[Win 7](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5b75e2fd6107b45e6d1eba45)

### CODE REVIEW

File: src/browser/bounds_changed_state_tracker.ts [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+import * as os from 'os';
+import * as Deferred from './deferred';
+import WindowGroups from './window_groups';
+import WindowGroupTransactionTracker, { ITransaction } from './window_group_transaction_tracker';
+import { toSafeInt } from '../common/safe_int';
+import ofEvents from './of_events';
+import route from '../common/route';
+import { clipBounds, windowSetBoundsToVisible } from './utils';
+import { OpenFinWindow, BrowserWindow } from '../shapes';
+import { Rectangle, windowTransaction } from 'electron';
```
Comment (HarsimranSingh) [inline src/browser/bounds_changed_state_tracker.ts]: What's the difference between this `windowTransaction` and the `WindowTransaction` on line 5

File: src/browser/bounds_changed_state_tracker.ts [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+    width: number;
+    height: number;
+}
+
+enum WindowState {
+    Normal = 'normal',
+    Maximized = 'maximized',
+    Minimized = 'minimized'
+}
+interface DefferedEvent {
```
Comment (HarsimranSingh) [inline src/browser/bounds_changed_state_tracker.ts]: :neckbeard: DeferredEvent

File: src/browser/bounds_changed_state_tracker.ts (hunk 1 above)
Comment (pbaize) [reply src/browser/bounds_changed_state_tracker.ts]: One is the typescript stub definition, the other is the js that actually runs. Having them not be the same isn't ideal, but I didn't want to change what other files already use.

**Comment by HarsimranSingh** [comment]
🌳 save for .36

**Pull request merged**